library/Businessprocess/BpNode.php

```diff
@@ -460,6 +462,16 @@ public function reCalculateState()
             case self::OP_OR:
                 $sort_state = min($sort_states);
                 break;
+            case self::OP_DEGRADED:
+                $maxState = max($sort_states);
+                $maxIcingaState = $maxState >> self::SHIFT_FLAGS;
+                $flags = $maxState & 0xf;
+
+                $warningState = ($this->stateToSortStateMap[self::ICINGA_WARNING] << self::SHIFT_FLAGS) + $flags;
```

Comment:
You should use `$this->getSortingState(self::WARNING)` here. It also maintains the flags already.

Reply:
I didn't update this due to the problem explain in the following comment.

Reply:
Gnarg, yes, this results in an infinite recursion. But I'd still use `$this->stateToSortState(self::ICINGA_WARNING)` instead of the direct map access.